Resources/public/js/views/services/ez-contenteditviewservice.js

```diff
@@ -40,24 +44,30 @@ YUI.add('ez-contenteditviewservice', function (Y) {
         _load: function (next) {
             var request = this.get('request'),
                 service = this,
-                languageCode = this.get('languageCode');
+                languageCode = this.get('languageCode'),
+                baseLanguageCode = this.get('baseLanguageCode'),
+                languageCodeForLoadContent = baseLanguageCode ? baseLanguageCode : languageCode;
 
             this.get('version').reset();
-            this._loadContent(request.params.id, languageCode, function () {
+            this._loadContent(request.params.id, languageCodeForLoadContent, function () {
                 var tasks,
-                    version = service.get('version'),
                     content = service.get('content'),
+                    fields = content.get('fields'),
+                    loadedContentLanguageCode = Y.Object.values(fields)[0].languageCode,
+                    setEmptyFields = false,
                     resources;
 
-                version.set('fields', content.get('fields'));
-
                 resources = content.get('resources');
 
+                if (!baseLanguageCode && loadedContentLanguageCode !== languageCode) {
+                    setEmptyFields = true;
```

Comment:
this is a bad name, *empty* is kind of wrong or at least inaccurate as the content type might define default value. I would rather call it `setDefaultFieldValue` and so the same kind of change everywhere you use the word *empty*.